### PR TITLE
 update swagger required fields

### PR DESCRIPTION
**Pull request by bogdan-rosianu**
## Reasoning
- There were some inconsistencies between the Swagger docs and the actual returned fields of some responses

## Proposed Changes
- Analyze the endpoint and introduce `required: false` for the Swagger APIProperty() decorator where needed
- Further fixes along the way

## How to test
-

### CODE REVIEW

File: src/endpoints/transactions/entities/transaction.operation.ts [hunk 1]
```diff
@@ -51,18 +51,18 @@ export class TransactionOperation {
...
 
-  @ApiProperty({ type: Number, nullable: true })
+  @ApiProperty({ type: Number, nullable: true , required: false })
   decimals?: number;
 
-  @ApiProperty({ type: String, nullable: true })
+  @ApiProperty({ type: String, nullable: true , required: false })
   data?: string;
 
   @ApiProperty()
```
Comment (cfaur09) [inline src/endpoints/transactions/entities/transaction.operation.ts:60]: missing required field

File: src/endpoints/transactions/entities/transaction.pool.ts [hunk 2]
```diff
@@ -5,27 +5,27 @@ export class TransactionPool {
     Object.assign(this, init);
   }
 
-  @ApiProperty({ type: String, description: 'Transaction hash', example: '39098e005c9f53622e9c8a946f9141d7c29a5da3bc38e07e056b549fa017ae1b' })
+  @ApiProperty({ type: String, description: 'Transaction hash', example: '39098e005c9f53622e9c8a946f9141d7c29a5da3bc38e07e056b549fa017ae1b' , required: false })
```
Comment (cfaur09) [inline src/endpoints/transactions/entities/transaction.pool.ts:8]: i think that for tx pool txHash, this field is required

File: src/endpoints/transactions/entities/transaction.pool.ts [hunk 3]
```diff
@@ -5,27 +5,27 @@ export class TransactionPool {
...
   txHash?: string;
 
-  @ApiProperty({ type: String, description: 'Sender bech32 address', example: 'erd1wh9c0sjr2xn8hzf02lwwcr4jk2s84tat9ud2kaq6zr7xzpvl9l5q8awmex' })
+  @ApiProperty({ type: String, description: 'Sender bech32 address', example: 'erd1wh9c0sjr2xn8hzf02lwwcr4jk2s84tat9ud2kaq6zr7xzpvl9l5q8awmex' , required: false })
```
Comment (cfaur09) [inline src/endpoints/transactions/entities/transaction.pool.ts:11]: same here

File: src/endpoints/accounts/entities/account.verification.ts [hunk 4]
```diff
@@ -7,15 +7,15 @@ export class AccountVerification {
     Object.assign(this, init);
   }
 
-  @ApiProperty({ description: 'Source code hash' })
+  @ApiProperty({ description: 'Source code hash' , required: false })
```
Comment (cfaur09) [inline src/endpoints/accounts/entities/account.verification.ts:10]: this entity is used only for account verification endpoint
codeHash should be required

File: src/endpoints/identities/entities/identity.ts [hunk 5]
```diff
@@ -6,31 +6,31 @@ export class Identity {
...
-  @ApiProperty({ type: String })
+  @ApiProperty({ type: String , required: false })
   location?: string;
 
-  @ApiProperty({ type: Number })
+  @ApiProperty({ type: Number , required: false })
   score?: number;
 
-  @ApiProperty({ type: Number })
+  @ApiProperty({ type: Number , required: false })
```
Comment (cfaur09) [inline src/endpoints/identities/entities/identity.ts:33]: validators should be required for identity

File: src/endpoints/nfts/entities/nft.ts [hunk 6]
```diff
@@ -20,7 +20,7 @@ export class Nft {
   @ApiProperty({ type: String })
   collection: string = '';
 
-  @ApiProperty({ type: Number, nullable: true })
+  @ApiProperty({ type: Number, nullable: true , required: false })
```
Comment (cfaur09) [inline src/endpoints/nfts/entities/nft.ts:23]: timestamp should be required for nft ( creation date )

File: src/endpoints/nfts/entities/nft.ts [hunk 7]
```diff
@@ -79,10 +79,10 @@ export class Nft {
   @ApiProperty({ type: Number, nullable: true })
   decimals: number | undefined = undefined;
 
-  @ApiProperty({ type: TokenAssets })
+  @ApiProperty({ type: TokenAssets , required: false })
   assets?: TokenAssets;
 
-  @ApiProperty({ type: String })
+  @ApiProperty({ type: String , required: false })
```
Comment (cfaur09) [inline src/endpoints/nfts/entities/nft.ts:85]: required

File: src/endpoints/nodes/entities/node.auction.ts [hunk 8]
```diff
@@ -17,10 +17,10 @@ export class NodeAuction {
   @ApiProperty({ type: String })
   avatar: string = '';
 
-  @ApiProperty({ type: String })
+  @ApiProperty({ type: String , required: false })
```
Comment (cfaur09) [inline src/endpoints/nodes/entities/node.auction.ts:20]: required

File: src/endpoints/nodes/entities/node.auction.ts [hunk 9]
```diff
@@ -17,10 +17,10 @@ export class NodeAuction {
...
   provider?: string = '';
 
-  @ApiProperty({ type: String })
+  @ApiProperty({ type: String , required: false })
```
Comment (cfaur09) [inline src/endpoints/nodes/entities/node.auction.ts:23]: required

File: src/endpoints/nfts/entities/nft.ts (hunk 6 above)
Comment (bogdan-rosianu) [reply src/endpoints/nfts/entities/nft.ts:23]: fixed

File: src/endpoints/nfts/entities/nft.ts (hunk 7 above)
Comment (bogdan-rosianu) [reply src/endpoints/nfts/entities/nft.ts:85]: fixed

File: src/endpoints/nodes/entities/node.auction.ts (hunk 8 above)
Comment (bogdan-rosianu) [reply src/endpoints/nodes/entities/node.auction.ts:20]: fixed

File: src/endpoints/nodes/entities/node.auction.ts (hunk 9 above)
Comment (bogdan-rosianu) [reply src/endpoints/nodes/entities/node.auction.ts:23]: fixed

File: src/endpoints/transactions/entities/transaction.operation.ts (hunk 1 above)
Comment (bogdan-rosianu) [reply src/endpoints/transactions/entities/transaction.operation.ts:60]: fixed

File: src/endpoints/transactions/entities/transaction.pool.ts (hunk 3 above)
Comment (bogdan-rosianu) [reply src/endpoints/transactions/entities/transaction.pool.ts:11]: fixed

File: src/endpoints/transactions/entities/transaction.pool.ts (hunk 2 above)
Comment (bogdan-rosianu) [reply src/endpoints/transactions/entities/transaction.pool.ts:8]: fixed

File: src/endpoints/transactions/entities/transaction.ts [hunk 10]
```diff
@@ -13,19 +13,19 @@ export class Transaction {
   @ApiProperty({ type: String })
   txHash: string = '';
 
-  @ApiProperty({ type: Number, nullable: true })
+  @ApiProperty({ type: Number, nullable: true, required: false })
```
Comment (cfaur09) [inline src/endpoints/transactions/entities/transaction.ts:16]: check transaction.ts, some fields should be required: true

**Review by cfaur09 — APPROVED** [review]

**Review by cfaur09 — APPROVED** [review]

**Pull request merged**